src/ode/native/native_generic.mli

```diff
@@ -50,43 +50,43 @@ module Make (M : Owl_types_ndarray_algodiff.Sig with type elt = float) : sig
     -> M.arr * float * float * bool
 
   val euler
-    : (module Types.SolverT
-         with type s = M.arr
-          and type t = M.arr
+    : (module Types.Solver
+         with type state = M.arr
+          and type f = M.arr -> float -> M.arr
           and type step_output = M.arr * float
-          and type output = M.arr * M.arr)
+          and type solve_output = M.arr * M.arr)
 
   val midpoint
-    : (module Types.SolverT
-         with type s = M.arr
-          and type t = M.arr
+    : (module Types.Solver
+         with type state = M.arr
+          and type f = M.arr -> float -> M.arr
           and type step_output = M.arr * float
-          and type output = M.arr * M.arr)
+          and type solve_output = M.arr * M.arr)
 
   val rk4
-    : (module Types.SolverT
-         with type s = M.arr
-          and type t = M.arr
+    : (module Types.Solver
+         with type state = M.arr
+          and type f = M.arr -> float -> M.arr
           and type step_output = M.arr * float
-          and type output = M.arr * M.arr)
+          and type solve_output = M.arr * M.arr)
 
   val rk23
     :  tol:float
     -> dtmax:float
-    -> (module Types.SolverT
-          with type s = M.arr
-           and type t = M.arr
+    -> (module Types.Solver
+          with type state = M.arr
+           and type f = M.arr -> float -> M.arr
            and type step_output = M.arr * float * float * bool
```

Comment:
We should really carefully document these... I have already forgotten what the additional fload and bool do.

Reply:
yea we should, I think the additional float and bool are new proposed `dt` and whether or not that step should be accepted